File: stripe/src/test/java/com/stripe/android/model/AccountParamsTest.java
package com.stripe.android.model;

import java.util.HashMap;
import java.util.Map;

import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

public class AccountParamsTest {

    @Test
    public void toParamMap_withBusinessData() {
        final Map<String, Object> businessData = new HashMap<>();
        businessData.put("name", "Stripe");
        final Map<String, Object> params =
                AccountParams.createAccountParams(true,
                        AccountParams.BusinessType.Company, businessData)
                        .toParamMap();

        //noinspection unchecked
        final Map<String, ?> accountData = (Map<String, ?>) params.get("account");
        assertNotNull(accountData);
        assertEquals(3, accountData.size());
        assertEquals(Boolean.TRUE, accountData.get(AccountParams.API_TOS_SHOWN_AND_ACCEPTED));
        assertEquals(AccountParams.BusinessType.Company.getCode(),
                accountData.get(AccountParams.API_BUSINESS_TYPE));
        assertEquals(businessData,
                accountData.get(AccountParams.BusinessType.Company.getCode()));
    }

    @Test
    public void toParamMap_withNoBusinessData() {
        final Map<String, Object> params =
                AccountParams.createAccountParams(true, null, null)
                        .toParamMap();

        //noinspection unchecked
        final Map<String, ?> accountData = (Map<String, ?>) params.get("account");
        assertNotNull(accountData);
        assertEquals(1, accountData.size());
        assertEquals(Boolean.TRUE, accountData.get(AccountParams.API_TOS_SHOWN_AND_ACCEPTED));
    }
}

File: stripe/src/test/java/com/stripe/android/model/AccountParamsTest.kt
package com.stripe.android.model

import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertNotNull

class AccountParamsTest {

    @Test
    fun toParamMap_withBusinessData() {
        val company = AccountParams.BusinessTypeParams.Company(name = "Stripe")
        val params = AccountParams.create(
            true,
            company
        )
            .toParamMap()

        val accountData = params["account"] as Map<String, *>?
        assertNotNull(accountData)
        assertEquals(3, accountData.size)
        assertEquals(java.lang.Boolean.TRUE, accountData[AccountParams.PARAM_TOS_SHOWN_AND_ACCEPTED])
        assertEquals(AccountParams.BusinessType.Company.code,
            accountData[AccountParams.PARAM_BUSINESS_TYPE])
        assertEquals(
            company.toParamMap(),
            accountData[AccountParams.BusinessType.Company.code]
        )
    }

    @Test
    fun toParamMap_withNoBusinessData() {
        val params = AccountParams.create(true).toParamMap()

        val accountData = params["account"] as Map<String, *>?
        assertNotNull(accountData)
        assertEquals(1, accountData.size)
        assertEquals(java.lang.Boolean.TRUE, accountData[AccountParams.PARAM_TOS_SHOWN_AND_ACCEPTED])
    }
}
